### PR TITLE
Review & Update

### PR DESCRIPTION
**Pull request by blakeneyops**
Mostly standardization with other OSS modules and some other small tweaks. Also added CI build.

### CODE REVIEW

File: certs/main.tf [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 }
 
-/* ---------------------------- */
-/* IAM Role & Instance Profile  */
-/* ---------------------------- */
-
+## Creates IAM role & policies
 resource "aws_iam_role" "vpn_role" {
-  name = "${var.region}-${var.stack_item_label}-vpn"
+  name = "${var.stack_item_label}-${var.region}"
```
Comment (blakeneyops) [inline certs/main.tf]: I prefer grouping by label than region.

File: certs/main.tf [hunk 2]
```diff
@@ -186,17 +164,16 @@ module "asg" {
   max_size         = 2
   min_size         = 1
   hc_grace_period  = 300
-  hc_check_type    = "EC2"
```
Comment (blakeneyops) [inline certs/main.tf]: Went with ELB health check

File: certs/main.tf [hunk 3]
```diff
@@ -214,11 +191,12 @@ resource "aws_elb" "elb" {
...
-    application = "${var.stack_item_label}-vpn"
+    Name        = "${var.stack_item_label}"
+    application = "${var.stack_item_fullname}"
     managed_by  = "terraform"
   }
-}
-
-# Create a Route53 record
 
+  lifecycle {
```
Comment (blakeneyops) [inline certs/main.tf]: This will address the dependency cycle I was observing

File: certs/templates/user_data.tpl [hunk 4]
```diff
@@ -1,12 +1,6 @@
 #cloud-config
-environment:
```
Comment (blakeneyops) [inline certs/templates/user_data.tpl]: Couldn't find any evidence in the AMI configuration that this data is used

File: certs/variables.tf [hunk 5]
```diff
@@ -1,66 +1,81 @@
...
-#
+###
+### 0 - if instance is a standalone instance outside a VPC
+### 1 - if instance is in a vpc
+###
 variable "in_vpc" {
-  default = 0
+  type        = "string"
+  description = "Flag for associating the cluster with a VPC."
+  default     = 1
```
Comment (blakeneyops) [inline certs/variables.tf]: Seemed to me VPC usage would be the default mode. Thoughts?

File: certs/variables.tf [hunk 6]
```diff
@@ -1,66 +1,81 @@
...
-
-#
-# m3.medium - if instance is a standalone instance outside a VPC
-# t2.small - if instance is in a vpc
-#
-variable "instance_type" {}
-
-variable "region" {}
-
-variable "release" {
```
Comment (blakeneyops) [inline certs/variables.tf]: Unused

File: certs/variables.tf [hunk 7]
```diff
@@ -1,66 +1,81 @@
...
-  default = "0.0.2"
+variable "instance_type" {
+  type        = "string"
+  description = "EC2 instance type to associate with the launch configuration."
+  default     = "t2.small"
 }
 
-variable "role" {
```
Comment (blakeneyops) [inline certs/variables.tf]: Unused

File: certs/variables.tf [hunk 8]
```diff
@@ -1,66 +1,81 @@
...
+  description = "Routes for the VPN server to expose"
+}
 
-variable "s3_bucket" {}
+variable "s3_bucket" {
+  type        = "string"
+  description = "The S3 bucket where certificate and configuration are stored."
+}
 
-variable "sns_topic_arn" {}
```
Comment (blakeneyops) [inline certs/variables.tf]: Unused

File: certs/main.tf (hunk 2 above)
Comment (yuhunglin) [inline certs/main.tf]: 👌

File: certs/main.tf (hunk 3 above)
Comment (yuhunglin) [inline certs/main.tf]: ah, lots o `create_before_destroy` bubbling up the stack.

File: certs/templates/user_data.tpl (hunk 4 above)
Comment (yuhunglin) [inline certs/templates/user_data.tpl]: 👌

File: certs/variables.tf (hunk 5 above)
Comment (yuhunglin) [inline certs/variables.tf]: Probably, we should probably check some defaults for the other elements and remove them if we do switch this default.

File: certs/variables.tf (hunk 6 above)
Comment (yuhunglin) [inline certs/variables.tf]: Hrmm, yeah this probably isn't the best place to track it. Wanted to somehow attach packer version numbers to the terraform builds. Any ideas?

File: certs/variables.tf (hunk 8 above)
Comment (yuhunglin) [inline certs/variables.tf]: 👍

File: certs/variables.tf (hunk 7 above)
Comment (yuhunglin) [inline certs/variables.tf]: 👍

**Comment by yuhunglin** [comment]
@blakeneyops looks good, I'll bump the release number with this merge so if there are issues i can roll back the module easily in legos.

File: certs/main.tf [hunk 9]
```diff
@@ -1,15 +1,8 @@
-# openvpn_server
-
-provider "aws" {
```
Comment (yuhunglin) [inline certs/main.tf]: @blakeneyops ah that's nice, the provider doesn't have to be in the module itself. Looks like switching to role based access won't be so bad after all.

**Pull request merged**